neqo-http3/src/features/extended_connect/connect_udp.rs

```diff
@@ -0,0 +1,422 @@
+// TODO: Rename to connect_udp_session.rs to be consistent with webtransport_session.rs?
+use std::{
+    cell::RefCell,
+    fmt::{self, Display, Formatter},
+    mem,
+    rc::Rc,
+};
+
+use neqo_common::{qtrace, Encoder, Header, MessageType, Role};
+use neqo_qpack as qpack;
+use neqo_transport::{Connection, DatagramTracking, StreamId};
+
+use crate::{
+    features::extended_connect::{ExtendedConnectEvents, ExtendedConnectType, SessionCloseReason}, priority::PriorityHandler, recv_message::{RecvMessage, RecvMessageInfo}, send_message::SendMessage, CloseType, Error, Http3StreamInfo, Http3StreamType, HttpRecvStream, HttpRecvStreamEvents, Priority, ReceiveOutput, RecvStream, RecvStreamEvents, Res, SendStream, SendStreamEvents, Stream
+};
+
+// TODO: De-duplicate with webtransport_session.rs?
+#[derive(Debug, PartialEq)]
+enum SessionState {
+    Negotiating,
+    Active,
+    FinPending,
+    Done,
+}
+
+impl SessionState {
+    pub const fn closing_state(&self) -> bool {
+        matches!(self, Self::FinPending | Self::Done)
+    }
+}
+
+#[derive(Debug)]
+pub struct ConnectUdpSession {
```

Comment:
Consider naming this:

```suggestion
pub struct Session {
```